src/test/java/spoon/test/type/TypeTest.java

```diff
@@ -174,8 +175,8 @@ public void testIntersectionTypeReferenceInGenericsAndCasts() throws Exception {
 		assertTrue(generic.getBoundingType() instanceof CtIntersectionTypeReference);
 		assertEquals("java.lang.Runnable & java.io.Serializable", generic.getBoundingType().toString());
 		final CtIntersectionTypeReference<?> superType = generic.getBoundingType().asCtIntersectionTypeReference();
-		assertEquals(aPozole.getFactory().Type().createReference(Runnable.class), superType.getBounds().get(0));
-		assertEquals(aPozole.getFactory().Type().createReference(Serializable.class), superType.getBounds().get(1));
+		assertEquals(aPozole.getFactory().Type().createReference(Runnable.class), superType.getBounds().stream().collect(Collectors.toList()).get(0));
```

Comment:
A comparator is used based on the `SourcePosition`: https://github.com/INRIA/spoon/pull/659/files#diff-9cd2517bedc2b7b0aba6d67ef583fee4R70